aws-s3-transfer-manager/src/operation/download/body.rs

```diff
@@ -50,9 +141,7 @@ impl Body {
     /// Returns [None] when there is no more data.
     /// Chunks returned from a [Body] are guaranteed to be sequenced
     /// in the right order.
-    pub async fn next(&mut self) -> Option<Result<AggregatedBytes, crate::error::Error>> {
-        // TODO(aws-sdk-rust#1159, design) - do we want ChunkResponse (or similar) rather than AggregatedBytes? Would
-        //  make additional retries of an individual chunk/part more feasible (though theoretically already exhausted retries)
+    pub async fn next(&mut self) -> Option<Result<ChunkResponse, crate::error::Error>> {
```

Comment:
One thing that comes to mind with this API is we only yield one chunk at a time when (now that we own the `AggregatedBytes` type) we could be yielding many chunks at once all stitched together into a single `AggregatedBytes`. 

I realize the point of the change we're making is to expose metadata per/chunk. I'm wondering though if we shouldn't also consider a batch recv API or an API that combines multiple chunks into one and returns as many as we have sequenced available. There is a possibility this makes a difference depending on how fast/slow a customer is processing data. Would have to benchmark I suppose. We can wait of course but it's worth bringing up for discussion now to see what people think and add TODO's maybe if we agree.

Reply:
That's pretty reasonable, IMO, something like [collect](https://docs.rs/aws-smithy-types/1.2.9/aws_smithy_types/byte_stream/struct.ByteStream.html#method.collect) in `ByteStream`.

Reply:
Thanks, yeah, it  makes sense, but we will need to benchmark it to see if there is any real-world performance gain. We should probably do it once we have a stricter global number-of-parts-in-memory scheduler in place because I think it will matter then. Currently, every download has its own queue, which is very large, so it might not matter much.